hack/build/bazel-docker.sh

```diff
@@ -0,0 +1,150 @@
+#See the License for the specific language governing permissions and
+#limitations under the License.
+
+script_dir="$(readlink -f $(dirname $0))"
+source "${script_dir}"/common.sh
+
+WORK_DIR="/go/src/kubevirt.io/containerized-data-importer"
+BUILDER_SPEC="${BUILD_DIR}/docker/builder"
+BUILDER_TAG="kubevirt-cdi-bazel-builder"
+BUILDER_VOLUME="kubevirt-cdi-volume"
+
+SYNC_OUT=${SYNC_OUT:-true}
+
+# Be less verbose with bazel
+if [ -n "${TRAVIS_JOB_ID}" ]; then
+    cat >.bazelrc <<EOF
+common --noshow_progress --noshow_loading_progress
+EOF
+fi
+
+# Build the encapsulated compile and test container
+(cd ${BUILDER_SPEC} && docker build --tag ${BUILDER_TAG} .)
+
+# Create the persistent docker volume
+if [ -z "$(docker volume list | grep ${BUILDER_VOLUME})" ]; then
+    docker volume create --name ${BUILDER_VOLUME}
+fi
+
+# Make sure that the output directory exists
+echo "Making sure output directory exists..."
+docker run -v "${BUILDER_VOLUME}:/root:rw,z" --security-opt label:disable --rm ${BUILDER_TAG} mkdir -p /root/go/src/kubevirt.io/containerized-data-importer/_out
+
+echo "Starting rsyncd"
+# Start an rsyncd instance and make sure it gets stopped after the script exits
+RSYNC_CID_CDI=$(docker run -d -v "${BUILDER_VOLUME}:/root:rw,z" --security-opt label:disable --expose 873 -P ${BUILDER_TAG} /usr/bin/rsync --no-detach --daemon --verbose)
+
+function finish() {
+    docker stop ${RSYNC_CID_CDI} >/dev/null 2>&1 &
+    docker rm -f ${RSYNC_CID_CDI} >/dev/null 2>&1 &
+}
+trap finish EXIT
+
+RSYNCD_PORT=$(docker port $RSYNC_CID_CDI 873 | cut -d':' -f2)
+
+rsynch_fail_count=0
+
+while ! rsync ${CDI_DIR}/${RSYNCTEMP} "rsync://root@127.0.0.1:${RSYNCD_PORT}/build/${RSYNCTEMP}" &>/dev/null; do
+    if [[ "$rsynch_fail_count" -eq 0 ]]; then
+        printf "Waiting for rsyncd to be ready"
+        sleep .1
+    elif [[ "$rsynch_fail_count" -lt 30 ]]; then
+        printf "."
+        sleep 1
+    else
+        printf "failed"
+        break
+    fi
+    rsynch_fail_count=$((rsynch_fail_count + 1))
+done
+
+printf "\n"
+
+rsynch_fail_count=0
+
+_rsync() {
+    rsync -al "$@"
+}
+
+echo "Rsyncing ${CDI_DIR} to container"
+# Copy kubevirt into the persistent docker volume
+_rsync \
+    --delete \
+    --exclude 'bazel-bin' \
+    --exclude 'bazel-genfiles' \
+    --exclude 'bazel-containerized-data-importer' \
+    --exclude 'bazel-out' \
+    --exclude 'bazel-testlogs' \
+    --exclude 'cluster-up/cluster/**/.kubectl' \
+    --exclude 'cluster-up/cluster/**/.oc' \
+    --exclude 'cluster-up/cluster/**/.kubeconfig' \
+    --exclude "_out" \
+    --exclude ".vagrant" \
+    ${CDI_DIR}/ \
+    "rsync://root@127.0.0.1:${RSYNCD_PORT}/build"
+
+volumes="-v ${BUILDER_VOLUME}:/root:rw,z"
+
+# append .docker directory as volume
+mkdir -p "${HOME}/.docker"
+volumes="$volumes -v ${HOME}/.docker:/root/.docker:ro,z"
+
+# Ensure that a bazel server is running
+
+if [ -z "$(docker ps --format '{{.Names}}' | grep ${BUILDER_VOLUME}-bazel-server)" ]; then
```

Comment:
maybe have a variable equal to `${BUILDER_VOLUME}-bazel-server`?

Reply:
Done